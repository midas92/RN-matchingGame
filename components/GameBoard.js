import React, { useState, useEffect } from 'react';
import { View, StyleSheet, Text } from 'react-native';
import GameTile from "./GameTile";

const GameBoard = ({
  gameMatrix,
  onTilePressed,
  difficulty
}) => {

  return (
    <View style={styles.boardContainer}>
      { gameMatrix.map((column, i) => {
        return (
          <View style={styles.matrixRow} key={i}>
            {
              column.map((tile, j) => {
                const { rowNumber, columnNumber, value, selected, locked } = tile;
                return (
                  <GameTile 
                    key={column.length * i + j} 
                    rowNumber={rowNumber}
                    columnNumber={columnNumber}
                    value={value}
                    selected={selected}
                    locked={locked}
                    onTilePressed={onTilePressed} 
                    totalColumns={column.length}
                    difficulty={difficulty}
                  />
                )
              })
            }
          </View>
        )
      })}
    </View>
  );
}

export default GameBoard;

const styles = StyleSheet.create({
  boardContainer: {
    margin: 10,
    backgroundColor: 'white',
    borderColor: 'white',
    borderWidth: 10,
    borderRadius: 10,
    shadowColor: 'black',
    shadowRadius: 10,
    shadowOpacity: 1,
  },
  matrixRow: {
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    margin: 3,
    backgroundColor: 'white',
    
  },
})