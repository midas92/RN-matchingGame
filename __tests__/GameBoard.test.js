import React from 'react';
import { render } from '@testing-library/react-native';
import GameBoard from '../components/GameBoard';

describe('<GameBoard/>', () => {
  let result;
  beforeEach(() => {
    const gameMatrix = [
      [ 
        { rowNumber: 0, columnNumber: 0, selected: false, locked: false, value: 1, }, 
        { rowNumber: 0, columnNumber: 1, selected: false, locked: false, value: 1, }, 
      ],
      [
        { rowNumber: 1, columnNumber: 0, selected: false, locked: false, value: 1 },
        { rowNumber: 1, columnNumber: 1, selected: false, locked: false, value: 1 },
      ]
    ];

    result = render(
      <GameBoard
        gameMatrix={gameMatrix}
        onTilePressed={jest.fn()}
        mode={'normal'}
      />
    );
  });

  it('should match snapshot', () => expect(result.toJSON()).toMatchSnapshot());
  it('should render the GameBoard component', () => expect(result).toBeTruthy());
});
