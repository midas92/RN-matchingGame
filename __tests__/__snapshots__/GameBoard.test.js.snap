// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameBoard/> should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "white",
      "borderRadius": 10,
      "borderWidth": 10,
      "margin": 10,
      "shadowColor": "black",
      "shadowOpacity": 1,
      "shadowRadius": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 3,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "backgroundColor": "gray",
            "borderRadius": "10%",
            "height": 150,
            "justifyContent": "center",
            "margin": 3,
            "shadowColor": "black",
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "transform": Array [
              Object {
                "rotateX": "0deg",
              },
            ],
            "width": 150,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "backgroundColor": "#34a8eb",
            "borderRadius": "10%",
            "height": 150,
            "justifyContent": "center",
            "margin": 3,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "transform": Array [
              Object {
                "rotateX": "180deg",
              },
            ],
            "width": 150,
          }
        }
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            style={
              Object {
                "alignItems": "center",
                "borderRadius": "10%",
                "height": 150,
                "justifyContent": "center",
                "margin": 3,
                "shadowColor": "black",
                "shadowOpacity": 1,
                "shadowRadius": 2,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 62.5,
                }
              }
            >
              1
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "backgroundColor": "gray",
            "borderRadius": "10%",
            "height": 150,
            "justifyContent": "center",
            "margin": 3,
            "shadowColor": "black",
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "transform": Array [
              Object {
                "rotateX": "0deg",
              },
            ],
            "width": 150,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "backgroundColor": "#34a8eb",
            "borderRadius": "10%",
            "height": 150,
            "justifyContent": "center",
            "margin": 3,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "transform": Array [
              Object {
                "rotateX": "180deg",
              },
            ],
            "width": 150,
          }
        }
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            style={
              Object {
                "alignItems": "center",
                "borderRadius": "10%",
                "height": 150,
                "justifyContent": "center",
                "margin": 3,
                "shadowColor": "black",
                "shadowOpacity": 1,
                "shadowRadius": 2,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 62.5,
                }
              }
            >
              1
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 3,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "backgroundColor": "gray",
            "borderRadius": "10%",
            "height": 150,
            "justifyContent": "center",
            "margin": 3,
            "shadowColor": "black",
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "transform": Array [
              Object {
                "rotateX": "0deg",
              },
            ],
            "width": 150,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "backgroundColor": "#34a8eb",
            "borderRadius": "10%",
            "height": 150,
            "justifyContent": "center",
            "margin": 3,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "transform": Array [
              Object {
                "rotateX": "180deg",
              },
            ],
            "width": 150,
          }
        }
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            style={
              Object {
                "alignItems": "center",
                "borderRadius": "10%",
                "height": 150,
                "justifyContent": "center",
                "margin": 3,
                "shadowColor": "black",
                "shadowOpacity": 1,
                "shadowRadius": 2,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 62.5,
                }
              }
            >
              1
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "backgroundColor": "gray",
            "borderRadius": "10%",
            "height": 150,
            "justifyContent": "center",
            "margin": 3,
            "shadowColor": "black",
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "transform": Array [
              Object {
                "rotateX": "0deg",
              },
            ],
            "width": 150,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "backgroundColor": "#34a8eb",
            "borderRadius": "10%",
            "height": 150,
            "justifyContent": "center",
            "margin": 3,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "transform": Array [
              Object {
                "rotateX": "180deg",
              },
            ],
            "width": 150,
          }
        }
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            style={
              Object {
                "alignItems": "center",
                "borderRadius": "10%",
                "height": 150,
                "justifyContent": "center",
                "margin": 3,
                "shadowColor": "black",
                "shadowOpacity": 1,
                "shadowRadius": 2,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 62.5,
                }
              }
            >
              1
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
  </View>
</View>
`;
